qualcommbe: ipq95xx: Add initial support for new target

Ansuel opened the pull request:
This PR adds initial support for IPQ95xx Qualcomm SoC.

Set target as source-only for now as only the Reference Board is
supported and needs to better evaluated with real Devices.

Qualcomm is making lots of effort in upstreaming all the bits
of the WiFi 7 platform and this will probably better supported
than WiFi 6 as it will have the PPE engine upstream supported
instead of relaying on NSS.

WiFi 7 support still needs to be correctly supported but also
OpenWrt still have a very early support for it.

Current version correctly support basic operation of the SoC
with Ethernet support.

Ansuel committed:
qualcommbe: ipq95xx: Add initial support for new target

Add initial support for new target with the initial patch for ethernet
support using pending upstream patches for PCS UNIPHY, PPE and EDMA.

Only initramfs currently working as support for new SPI/NAND
implementation, USB, CPUFreq and other devices is still unfinished and
needs to be evaluated.

Link: https://github.com/openwrt/openwrt/pull/17725
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
qualcommbe: ipq95xx: Backport cpufreq patch and enable RPM

Backport CPUFreq patch and enable RPM. This is to enable CPU Frequency
scaling and regulators.

Link: https://github.com/openwrt/openwrt/pull/17725
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
qualcommbe: ipq95xx: Add patch fixing USB regulator

Add patch fixing USB regulator as the LDO needs to use ID 5 instead of
ID 2.

Link: https://github.com/openwrt/openwrt/pull/17725
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
qualcommbe: ipq95xx: Add patch adding partition table common to RDP

Add patches adding partition table common to RDP board and node for AQR
NVMEM.

Link: https://github.com/openwrt/openwrt/pull/17725
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
qualcommbe: ipq95xx: Add patch fixing unmet dependency for QCOM PPE

Add patch fixing unmet dependency for QCOM PPE. This fix a compilation
error when SFP config is selected.

Link: https://github.com/openwrt/openwrt/pull/17725
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
qualcommbe: ipq95xx: Update patch with upstream version and refresh

Update patch with upstream version and automatically refresh with make
target/linux/refresh.

Also backport one additional fix patch for NAND patch and drop a patch
merged upstream.

Link: https://github.com/openwrt/openwrt/pull/17725
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
qualcommbe: ipq95xx: Enable Aquantia PHY by default

Enable the Aquantia PHY by default in config-defaults. Target is big
enough to permit embedding the PHY instead of having as an external
module.

Link: https://github.com/openwrt/openwrt/pull/17725
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
qualcommbe: ipq95xx: Set target as source-only

Set target as source-only for now as only the Reference Board is
supported and needs to better evaluated with real Devices.

Link: https://github.com/openwrt/openwrt/pull/17725
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

wigyori commented:
Hi - can you show us a bootlog on where you are with the support for the board and testing the interfaces ?

Ansuel commented:
@wigyori yes I will provide it. Currently the board fully boots and all the ESS port comes up and works. Notice that there is currently a problem with RPM where some old version cause the port not working if the RPM is enabled in DTS. Qcom is aware and I hope we will find a solution for it.

Ansuel commented:
```
[    0.000000] Booting Linux on physical CPU 0x0000000000 [0x411fd090]
[    0.000000] Linux version 6.6.71 (ansuel@Ansuel-XPS24) (aarch64-openwrt-linux-musl-gcc (OpenWrt GCC 13.3.0 r0+27995-57f2c72505) 13.3.0, GNU ld (GNU Binutils) 2.42) #0 SMP Sat Jan 25 20:19:10 2025
[    0.000000] Machine model: Qualcomm Technologies, Inc. IPQ9574/AP-AL02-C7
[    0.000000] OF: reserved mem: 0x000000004a100000..0x000000004a4fffff (4096 KiB) nomap non-reusable bootloader@4a100000
[    0.000000] OF: reserved mem: 0x000000004a500000..0x000000004a5fffff (1024 KiB) nomap non-reusable sbl@4a500000
[    0.000000] OF: reserved mem: 0x000000004a600000..0x000000004a9fffff (4096 KiB) nomap non-reusable tz@4a600000
[    0.000000] OF: reserved mem: 0x000000004aa00000..0x000000004aafffff (1024 KiB) nomap non-reusable smem@4aa00000
[    0.000000] Zone ranges:
[    0.000000]   DMA      [mem 0x0000000040000000-0x000000007fffffff]
[    0.000000]   DMA32    empty
[    0.000000]   Normal   empty
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000040000000-0x000000004a0fffff]
[    0.000000]   node   0: [mem 0x000000004a100000-0x000000004aafffff]
[    0.000000]   node   0: [mem 0x000000004ab00000-0x000000007fffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000040000000-0x000000007fffffff]
[    0.000000] psci: probing for conduit method from DT.
[    0.000000] psci: PSCIv1.0 detected in firmware.
[    0.000000] psci: Using standard PSCI v0.2 function IDs
[    0.000000] psci: MIGRATE_INFO_TYPE not supported.
[    0.000000] psci: SMC Calling Convention v1.0
[    0.000000] percpu: Embedded 18 pages/cpu s35496 r8192 d30040 u73728
[    0.000000] Detected VIPT I-cache on CPU0
[    0.000000] CPU features: detected: Spectre-v2
[    0.000000] CPU features: detected: Spectre-v4
[    0.000000] CPU features: detected: Spectre-BHB
[    0.000000] alternatives: applying boot alternatives
[    0.000000] Kernel command line: console=ttyMSM0,115200n8 ubi.mtd=rootfs root=mtd:ubi_rootfs rootfstype=squashfs
[    0.000000] Dentry cache hash table entries: 131072 (order: 8, 1048576 bytes, linear)
[    0.000000] Inode-cache hash table entries: 65536 (order: 7, 524288 bytes, linear)
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 258048
[    0.000000] mem auto-init: stack:off, heap alloc:off, heap free:off
[    0.000000] software IO TLB: SWIOTLB bounce buffer size adjusted to 1MB
[    0.000000] software IO TLB: area num 4.
[    0.000000] software IO TLB: mapped [mem 0x000000007ea80000-0x000000007eb80000] (1MB)
[    0.000000] Memory: 996944K/1048576K available (8448K kernel code, 944K rwdata, 2692K rodata, 7360K init, 279K bss, 51632K reserved, 0K cma-reserved)
[    0.000000] SLUB: HWalign=64, Order=0-3, MinObjects=0, CPUs=4, Nodes=1
[    0.000000] rcu: Hierarchical RCU implementation.
[    0.000000]  Tracing variant of Tasks RCU enabled.
[    0.000000] rcu: RCU calculated value of scheduler-enlistment delay is 10 jiffies.
[    0.000000] NR_IRQS: 64, nr_irqs: 64, preallocated irqs: 0
[    0.000000] Root IRQ handler: gic_handle_irq
[    0.000000] GICv2m: range[mem 0x0b00c000-0x0b00cffc], SPI[640:671]
[    0.000000] GICv2m: range[mem 0x0b00d000-0x0b00dffc], SPI[672:703]
[    0.000000] GICv2m: range[mem 0x0b00e000-0x0b00effc], SPI[704:735]
[    0.000000] rcu: srcu_init: Setting srcu_struct sizes based on contention.
[    0.000000] arch_timer: cp15 and mmio timer(s) running at 24.00MHz (virt/virt).
[    0.000000] clocksource: arch_sys_counter: mask: 0xffffffffffffff max_cycles: 0x588fe9dc0, max_idle_ns: 440795202592 ns
[    0.000000] sched_clock: 56 bits at 24MHz, resolution 41ns, wraps every 4398046511097ns
[    0.000046] Calibrating delay loop (skipped), value calculated using timer frequency.. 48.00 BogoMIPS (lpj=240000)
[    0.000056] pid_max: default: 32768 minimum: 301
[    0.003985] Mount-cache hash table entries: 2048 (order: 2, 16384 bytes, linear)
[    0.003995] Mountpoint-cache hash table entries: 2048 (order: 2, 16384 bytes, linear)
[    0.007033] RCU Tasks Trace: Setting shift to 2 and lim to 1 rcu_task_cb_adjust=1 rcu_task_cpu_ids=4.
[    0.007160] rcu: Hierarchical SRCU implementation.
[    0.007163] rcu:     Max phase no-delay instances is 1000.
[    0.007600] smp: Bringing up secondary CPUs ...
[    0.009041] Detected VIPT I-cache on CPU1
[    0.009127] CPU1: Booted secondary processor 0x0000000001 [0x411fd090]
[    0.010727] Detected VIPT I-cache on CPU2
[    0.010812] CPU2: Booted secondary processor 0x0000000002 [0x411fd090]
[    0.012420] Detected VIPT I-cache on CPU3
[    0.012503] CPU3: Booted secondary processor 0x0000000003 [0x411fd090]
[    0.012591] smp: Brought up 1 node, 4 CPUs
[    0.012598] SMP: Total of 4 processors activated.
[    0.012602] CPU features: detected: 32-bit EL0 Support
[    0.012606] CPU features: detected: CRC32 instructions
[    0.012661] CPU features: emulated: Privileged Access Never (PAN) using TTBR0_EL1 switching
[    0.012666] CPU: All CPU(s) started at EL1
[    0.012668] alternatives: applying system-wide alternatives
[    0.016908] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.016921] futex hash table entries: 1024 (order: 4, 65536 bytes, linear)
[    0.018355] pinctrl core: initialized pinctrl subsystem
[    0.019827] NET: Registered PF_NETLINK/PF_ROUTE protocol family
[    0.020557] DMA: preallocated 128 KiB GFP_KERNEL pool for atomic allocations
[    0.020612] DMA: preallocated 128 KiB GFP_KERNEL|GFP_DMA pool for atomic allocations
[    0.020665] DMA: preallocated 128 KiB GFP_KERNEL|GFP_DMA32 pool for atomic allocations
[    0.020921] thermal_sys: Registered thermal governor 'step_wise'
[    0.020959] cpuidle: using governor menu
[    0.021081] ASID allocator initialised with 65536 entries
[    0.024165] /soc@0/interrupt-controller@b000000: Fixed dependency cycle(s) with /soc@0/interrupt-controller@b000000
[    0.034372] Modules: 27792 pages in range for non-PLT usage
[    0.034379] Modules: 519312 pages in range for PLT usage
[    0.038439] SCSI subsystem initialized
[    0.038567] usbcore: registered new interface driver usbfs
[    0.038597] usbcore: registered new interface driver hub
[    0.038621] usbcore: registered new device driver usb
[    0.039338] qcom_scm: convention: smc arm 64
[    0.040269] s1: Bringing 0uV into 725000-725000uV
[    0.040417] qcom_rpm_smd_regulator remoteproc:glink-edge:rpm-requests:regulators: Supply for s1 (s1) resolved to itself
[    0.040501] l2: Bringing 0uV into 1800000-1800000uV
[    0.040927] clocksource: Switched to clocksource arch_sys_counter
[    0.044029] NET: Registered PF_INET protocol family
[    0.044138] IP idents hash table entries: 16384 (order: 5, 131072 bytes, linear)
[    0.046251] tcp_listen_portaddr_hash hash table entries: 512 (order: 1, 8192 bytes, linear)
[    0.046267] Table-perturb hash table entries: 65536 (order: 6, 262144 bytes, linear)
[    0.046279] TCP established hash table entries: 8192 (order: 4, 65536 bytes, linear)
[    0.046329] TCP bind hash table entries: 8192 (order: 6, 262144 bytes, linear)
[    0.046522] TCP: Hash tables configured (established 8192 bind 8192)
[    0.046905] MPTCP token hash table entries: 1024 (order: 2, 24576 bytes, linear)
[    0.047041] UDP hash table entries: 512 (order: 2, 16384 bytes, linear)
[    0.047068] UDP-Lite hash table entries: 512 (order: 2, 16384 bytes, linear)
[    0.047424] NET: Registered PF_UNIX/PF_LOCAL protocol family
[    0.047452] PCI: CLS 0 bytes, default 64
[    0.052291] workingset: timestamp_bits=46 max_order=18 bucket_order=0
[    0.052727] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.052731] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    0.059093] qcom-qusb2-phy 7b000.phy: Registered Qcom-QUSB2 phy
[    0.068754] bam-dma-engine 704000.dma-controller: num-channels unspecified in dt
[    0.068766] bam-dma-engine 704000.dma-controller: num-ees unspecified in dt
[    0.071825] Serial: 8250/16550 driver, 2 ports, IRQ sharing disabled
[    0.072413] msm_serial 78b1000.serial: msm_serial: detected port #0
[    0.072437] msm_serial 78b1000.serial: uartclk = 1843199
[    0.072658] 78b1000.serial: ttyMSM0 at MMIO 0x78b1000 (irq = 21, base_baud = 115199) is a MSM
[    0.072680] msm_serial: console setup on port #0
[    0.072725] printk: console [ttyMSM0] enabled
[    0.811620] msm_serial: driver initialized
[    0.819159] loop: module loaded
[    0.820396] spi_qup 78b5000.spi: IN:block:16, fifo:64, OUT:block:16, fifo:64
[    0.823197] spi-nor spi0.0: mx25u25635f (32768 Kbytes)
[    2.055061] 21 fixed-partitions partitions found on MTD device spi0.0
[    2.055342] Creating 21 MTD partitions on "spi0.0":
[    2.060482] 0x000000000000-0x0000000c0000 : "0:sbl1"
[    2.065602] 0x0000000c0000-0x0000000d0000 : "0:mibib"
[    2.070695] 0x0000000d0000-0x0000000f0000 : "0:bootconfig"
[    2.075627] 0x0000000f0000-0x000000110000 : "0:bootconfig1"
[    2.081125] 0x000000110000-0x000000290000 : "0:qsee"
[    2.086483] 0x000000290000-0x000000410000 : "0:qsee_1"
[    2.091679] 0x000000410000-0x000000420000 : "0:devcfg"
[    2.096631] 0x000000420000-0x000000430000 : "0:devcfg_1"
[    2.102370] 0x000000430000-0x000000440000 : "0:apdp"
[    2.107233] 0x000000440000-0x000000450000 : "0:apdp_1"
[    2.112217] 0x000000450000-0x000000490000 : "0:tme"
[    2.117122] 0x000000490000-0x0000004d0000 : "0:tme_1"
[    2.121936] 0x0000004d0000-0x0000004f0000 : "0:rpm"
[    2.127082] 0x0000004f0000-0x000000510000 : "0:rpm_1"
[    2.131853] 0x000000510000-0x000000520000 : "0:cdt"
[    2.136988] 0x000000520000-0x000000530000 : "0:cdt_1"
[    2.141687] 0x000000530000-0x000000540000 : "0:appsblenv"
[    2.146832] 0x000000540000-0x0000005e0000 : "0:appsbl"
[    2.152323] 0x0000005e0000-0x000000680000 : "0:appsbl_1"
[    2.157296] 0x000000680000-0x000000780000 : "0:art"
[    2.162775] 0x000000780000-0x000000800000 : "0:ethphyfw"
[    2.522475] dummy# (dummy): netif_napi_add_weight() called with weight 128
[    2.524053] qcom_ppe 3a000000.ethernet: EDMA configuration successful
[    2.528365] qcom_ppe 3a000000.ethernet lan1 (uninitialized): GMAC1 Using random MAC address - 46:f1:10:f2:23:09
[    2.731563] qcom_ppe 3a000000.ethernet lan1 (uninitialized): PHY [90000.mdio-1:10] driver [Qualcomm QCA8075] (irq=POLL)
[    2.732221] qcom_ppe 3a000000.ethernet lan2 (uninitialized): GMAC2 Using random MAC address - 62:4e:32:bb:69:fe
[    2.821217] qcom_ppe 3a000000.ethernet lan2 (uninitialized): PHY [90000.mdio-1:11] driver [Qualcomm QCA8075] (irq=POLL)
[    2.821818] qcom_ppe 3a000000.ethernet lan3 (uninitialized): GMAC3 Using random MAC address - 6a:9c:7b:99:49:84
[    2.911216] qcom_ppe 3a000000.ethernet lan3 (uninitialized): PHY [90000.mdio-1:12] driver [Qualcomm QCA8075] (irq=POLL)
[    2.911805] qcom_ppe 3a000000.ethernet lan4 (uninitialized): GMAC4 Using random MAC address - f6:91:83:0d:ec:e9
[    3.001212] qcom_ppe 3a000000.ethernet lan4 (uninitialized): PHY [90000.mdio-1:13] driver [Qualcomm QCA8075] (irq=POLL)
[    3.001891] qcom_ppe 3a000000.ethernet lan5 (uninitialized): GMAC5 Using random MAC address - 1e:5e:4b:83:3e:6f
[    3.021025] qcom_ppe 3a000000.ethernet lan5 (uninitialized): PHY [90000.mdio-1:08] driver [Aquantia AQR113C] (irq=POLL)
[    3.021622] qcom_ppe 3a000000.ethernet wan (uninitialized): GMAC6 Using random MAC address - c6:08:77:d6:92:2f
[    3.041028] qcom_ppe 3a000000.ethernet wan (uninitialized): PHY [90000.mdio-1:00] driver [Aquantia AQR113C] (irq=POLL)
[    3.042728] i2c_dev: i2c /dev entries driver
[    3.056282] sdhci: Secure Digital Host Controller Interface driver
[    3.056795] sdhci: Copyright(c) Pierre Ossman
[    3.062821] sdhci-pltfm: SDHCI platform and OF driver helper
[    3.085004] NET: Registered PF_INET6 protocol family
[    3.086326] Segment Routing with IPv6
[    3.089053] In-situ OAM (IOAM) with IPv6
[    3.092670] NET: Registered PF_PACKET protocol family
[    3.096696] 8021q: 802.1Q VLAN Support v1.8
[    3.130435] cpufreq: cpufreq_online: CPU0: Running at unlisted initial frequency: 24000 KHz, changing to: 936000 KHz
[    3.131263] UBI error: cannot open mtd rootfs, [    3.146523] Freeing unused kernel memory: 7360K
[    3.146585] Run /init as init process
[    3.206157] init: Console is alive
[    3.206231] init: - watchdog -
[    3.210653] kmodloader: loading kernel modules from /etc/modules-boot.d/*
[    3.212345] gpio_button_hotplug: loading out-of-tree module taints kernel.
[    3.219056] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
[    3.228817] init: - preinit -
[   12.080925] random: crng init done
[   12.134874] qcom_ppe 3a000000.ethernet lan1: configuring for inband/qsgmii link mode
Press the [f] key and hit [enter] to enter failsafe mode
Press the [1], [2], [3] or [4] key and hit [enter] to select the debug level
[   13.270977] qcom,gcc-ipq9574 1800000.clock-controller: sync_state() pending due to 79b0000.spi
[   13.271009] qcom,gcc-ipq9574 1800000.clock-controller: sync_state() pending due to 8af8800.usb
[   16.403848] procd: - early -
[   16.403918] procd: - watchdog -
[   16.923523] procd: - watchdog -
[   16.923740] procd: - ubus -
[   16.976047] procd: - init -
Please press Enter to activate this console.
[   17.047599] kmodloader: loading kernel modules from /etc/modules.d/*
[   17.086185] jitterentropy: Initialization failed with host not compliant with requirements: 9
[   17.123304] urngd: v1.0.2 started.
[   17.194544] PPP generic driver version 2.4.2
[   17.194925] NET: Registered PF_PPPOX protocol family
[   17.217928] kmodloader: done loading kernel modules from /etc/modules.d/*



BusyBox v1.36.1 (2024-12-13 11:20:36 UTC) built-in shell (ash)

  _______                     ________        __
 |       |.-----.-----.-----.|  |  |  |.----.|  |_
 |   -   ||  _  |  -__|     ||  |  |  ||   _||   _|
 |_______||   __|_____|__|__||________||__|  |____|
          |__| W I R E L E S S   F R E E D O M
 -----------------------------------------------------
 OpenWrt SNAPSHOT, r0+28124-fc4b345ee3
 -----------------------------------------------------
=== WARNING! =====================================
There is no root password defined on this device!
Use the "passwd" command to set up a new password
in order to prevent unauthorized SSH logins.
--------------------------------------------------
root@OpenWrt:~# [   20.514086] qcom_ppe 3a000000.ethernet lan1: configuring for inband/qsgmii link mode
[   20.681334] br-lan: port 1(lan1) entered blocking state
[   20.681364] br-lan: port 1(lan1) entered disabled state
[   20.685375] qcom_ppe 3a000000.ethernet lan1: entered allmulticast mode
[   20.690698] qcom_ppe 3a000000.ethernet lan1: entered promiscuous mode
[   20.701692] qcom_ppe 3a000000.ethernet lan2: configuring for inband/qsgmii link mode
[   20.871422] br-lan: port 2(lan2) entered blocking state
[   20.871452] br-lan: port 2(lan2) entered disabled state
[   20.875464] qcom_ppe 3a000000.ethernet lan2: entered allmulticast mode
[   20.880831] qcom_ppe 3a000000.ethernet lan2: entered promiscuous mode
[   20.889414] qcom_ppe 3a000000.ethernet lan3: configuring for inband/qsgmii link mode
[   21.061324] br-lan: port 3(lan3) entered blocking state
[   21.061347] br-lan: port 3(lan3) entered disabled state
[   21.065361] qcom_ppe 3a000000.ethernet lan3: entered allmulticast mode
[   21.070722] qcom_ppe 3a000000.ethernet lan3: entered promiscuous mode
[   21.079465] qcom_ppe 3a000000.ethernet lan4: configuring for inband/qsgmii link mode
[   21.251300] br-lan: port 4(lan4) entered blocking state
[   21.251324] br-lan: port 4(lan4) entered disabled state
[   21.255336] qcom_ppe 3a000000.ethernet lan4: entered allmulticast mode
[   21.260674] qcom_ppe 3a000000.ethernet lan4: entered promiscuous mode
[   21.269577] qcom_ppe 3a000000.ethernet lan5: configuring for inband/usxgmii link mode
[   21.301355] br-lan: port 5(lan5) entered blocking state
[   21.301377] br-lan: port 5(lan5) entered disabled state
[   21.305390] qcom_ppe 3a000000.ethernet lan5: entered allmulticast mode
[   21.310754] qcom_ppe 3a000000.ethernet lan5: entered promiscuous mode
[   21.320113] qcom_ppe 3a000000.ethernet wan: configuring for inband/usxgmii link mode
```

This is the current bootlog

wigyori commented:
Thanks for the bootlog. If you say the ethernet ports work fine, that's great.

I still don't get why a "source-only" target has been merged, even if the IPQ95xx is a nice platform. I recalled that merging any new targets require that a large portion of the peripherals should work (ethernet, OK, you seem to have that, but I can't see a link up or a ping test), but there is no PCIe, can't see USB, and so on. I and probably others would have a ton of new/shiny stuff to be merged this way, but aren't coming forward with it, either for waiting until patches published to bring the peripherals up, or working on actual patches to bring these up - prior to merging the target.

Ansuel commented:
@wigyori Link up and ping works, I wouldn't merge a device that doesn't have at least the ethernet part working. Sorry if premature merge caused any problem, it's really to have a starting point to identify all the merged patch and further progress with tracking of this platform.

HeyMeco commented:
The only device I know of that also uses this is the Alta Labs Route10. Might be a solid option to get OpenWRT running on as it runs the qualcomm fork under the hood

fightforlife commented:
From Google and the GPL code I found the Mercursys MR47BE should also use this chip. This sound like a relatively interesting option since it is a cheap (150€) triband 2x2 router. I am inclined to get this device as a project.

PerezChilli commented:
> The only device I know of that also uses this is the Alta Labs Route10. Might be a solid option to get OpenWRT running on as it runs the qualcomm fork under the hood

Examples

Xiaomi BE7000 (ipq9554), Xiaomi BE10000 (ipq9570), Linksys Velop WRT Pro 7 (ipq9554)

wigyori commented:
> From Google and the GPL code I found the Mercursys MR47BE should also use this chip. This sound like a relatively interesting option since it is a cheap (150€) triband 2x2 router. I am inclined to get this device as a project.

Make sure you check first that the QCA8084 PHY is supported by the qcom_ppe driver - the RDP433 devboard has QCA8075 (1G), while the MR47BE has QCA8084 (2.5G).

castillofrancodamian commented:
> > The only device I know of that also uses this is the Alta Labs Route10. Might be a solid option to get OpenWRT running on as it runs the qualcomm fork under the hood
> 
> Examples
> 
> Xiaomi BE7000 (ipq9554), Xiaomi BE10000 (ipq9570), Linksys Velop WRT Pro 7 (ipq9554)

Deco BE85 and BE95 too. More devices [here](https://www.qualcomm.com/products/internet-of-things/networking/wi-fi-networks/networking-device-finder).

mantas-p commented on target/linux/qualcommbe/Makefile:
```diff
@@ -0,0 +1,20 @@
+BOARDNAME:=Qualcomm Atheros 802.11be WiSoC-s
+FEATURES:=squashfs ramdisk fpu nand rtc emmc source-only
+KERNELNAME:=Image
+CPU_TYPE:=cortex-a53
```
IPQ95xx is cortex-a73

Ansuel replied:
As said in the merged commit. perf gain is in the order of 1x with the cons of having to build all packages for a72 so for now we are using a53

AEGEGE commented:
How far have you come so far? How long will it take to be available?

Ansuel commented:
@AEGEGE what do you mean?

AEGEGE commented:
> @AEGEGE what do you mean?

I mean when can openwrt be used stably on Xiaomi BE10000?

Ansuel commented:
I mean depends on what the device have. Also is it even possible to flash it?

AEGEGE commented:
The device can be flashed with openwrt, and the hardware configuration is as follows:
SoC : QCOM IPQ9570
RAM : 2GB DDR4
Flash : 256MB
10G PHY: AQR113C 
2.5G PHY: QCA8084
WiFi : 1 x 2.4GHz  QCN6214
2 x 5GHz QCN6274

Ansuel commented:
Ok so only 8084 is currently not working and needs some love.

Ansuel commented:
The fem controller should be out of scope. For qcn in theory they should work are they are handled by ath12k.

OodavidsinoO commented:
> I mean depends on what the device have. Also is it even possible to flash it?

Yes, but need a bit of hacking. As long as you downgrade back to v1.0.53 firmware, you can use the broken access control of the internal Docker to gain root, do root persisting and flash the U-Boot.

English Writeup: [XiaoMi BE10000 SSH Unlock with root](https://www-gaicas-com.translate.goog/xiaomi-be10000.html?_x_tr_sl=auto&_x_tr_tl=zh-CN&_x_tr_hl=en-US&_x_tr_pto=wapp)
Original Writeup: [小米万兆路由器 解锁SSH 启用科学上网](https://www.gaicas.com/xiaomi-be10000.html)

AEGEGE commented:
> > I mean depends on what the device have. Also is it even possible to flash it?
> 
> Permanent ssh access can be obtained through "xmir-patcher"

Regarding another router tplink 7TR 13090 BE13000, is there any relevant material for this one?

artur131 commented:
> > > I mean depends on what the device have. Also is it even possible to flash it?
> > 
> > 
> > Permanent ssh access can be obtained through "xmir-patcher"
> 
> Regarding another router tplink 7TR 13090 BE13000, is there any relevant material for this one?

I am not familiar with this device

peterwillcn commented:
Can it support Xiaomi 10000 ?
link: 
https://forum.openwrt.org/t/qualcommbe-ipq95xx-add-xiaomi-10000-support/242655
https://forum.openwrt.org/t/openwrt-support-for-xiaomi-ax10000/161813
